src/model/formats/foreflight-writer.ts

```diff
@@ -0,0 +1,97 @@
+import { ChecklistFile, ChecklistGroup, ChecklistItem, ChecklistItem_Type } from '../../../gen/ts/checklist';
+import {
+  ForeFlightChecklistContainer,
+  ForeFlightChecklistGroup,
+  ForeFlightChecklistItem,
+} from '../../../gen/ts/foreflight';
+
+import { ForeFlightCategory, ForeFlightUtils } from './foreflight-utils';
+
+export class ForeFlightWriter {
+  public static async write(file: ChecklistFile): Promise<Blob> {
+    return ForeFlightUtils.encrypt(
+      ForeFlightChecklistContainer.toJsonString(await ForeFlightWriter.checklistFileToFF(file), { prettySpaces: 2 }),
+    );
+  }
+
+  private static async checklistFileToFF(file: ChecklistFile): Promise<ForeFlightChecklistContainer> {
+    return {
+      type: ForeFlightUtils.CONTAINER_TYPE,
+      payload: {
+        objectId: await ForeFlightUtils.getObjectId(),
+        schemaVersion: ForeFlightUtils.SCHEMA_VERSION,
+        metadata: {
+          name: file.metadata?.name,
+          detail: file.metadata?.description,
+          tailNumber: file.metadata?.aircraftInfo,
+        },
+        groups: await ForeFlightWriter.checklistGroupsToFF(file.groups),
+      },
+    };
+  }
+
+  private static async checklistGroupsToFF(groupsEFIS: ChecklistGroup[]): Promise<ForeFlightChecklistGroup[]> {
+    return Promise.all(
+      Object.values(ForeFlightCategory).map(
+        async (categoryFF): Promise<ForeFlightChecklistGroup> => ({
+          objectId: await ForeFlightUtils.getObjectId(),
+          groupType: categoryFF,
```

Comment:
Nitpicking: call it either category or group type uniformly

Reply:
With your latest change for object IDs, you should be able to remove *all* async and aawait from these functions (except write() probably)

Reply:
Thanks, I naively expected the linter to warn me about this, so I missed it. Makes everything much easier! Fixed in 99b41d2.